analyses/CLK1-splicing_correlations/10-plot-clk1-ex4-transcripts-hgg-normals.R

```diff
@@ -0,0 +1,319 @@
+      grepl("PONS", aliquot_id, ignore.case = TRUE) ~ "Pons",
+      grepl("frontal", aliquot_id, ignore.case = TRUE) ~ "Frontal_Cortex",
+      TRUE ~ NA_character_  # Default to NA if no match
+    )
+  ) %>%
+  dplyr::select(-aliquot_id) 
+  
+# ENST00000434813.3, ENST00000321356.9, ENST00000409403.6, ENST00000434813.3
+ped_clk1_transcr_counts <- readRDS(ped_trans_file) %>%
+  filter(grepl("^CLK1", gene_symbol)) %>%
+  mutate(
+    transcript_id = case_when(
+      transcript_id %in% c("ENST00000321356.9","ENST00000434813.3", "ENST00000409403.6") ~ "Exon 4",  # Rename specified transcripts
+      TRUE ~ "Other"  # All other transcripts are renamed to "Other"
+    ) ) %>%
+  group_by(transcript_id) %>%
+  summarise(across(starts_with("SRR"), sum, na.rm = TRUE)) %>%
+  #select(-gene_symbol) %>% # Remove 'gene_symbol'
+  pivot_longer(
+    cols = -transcript_id,
+    names_to = "Sample",
+    values_to = "TPM"
+  ) %>%
+  dplyr::mutate(group="Pediatric normals") %>%
+  dplyr::rename(Kids_First_Biospecimen_ID=Sample) %>%
+  inner_join(sra_mapping,by=c("Kids_First_Biospecimen_ID" = "SRR"))
+
+transcript_expr_CLK1_combined_df <- rbind(all_clk4_transcr_counts,ped_clk1_transcr_counts,gtex_clk1_transc_counts,evodevo_clk1_transc_counts,clk4_transcr_counts_astro) %>% 
+  group_by(Kids_First_Biospecimen_ID) %>%
+  mutate(total_TPM = sum(TPM[transcript_id %in% c("Exon 4", "Other")], na.rm = TRUE)) %>%
+  mutate(proportion = ifelse(transcript_id == "Exon 4", TPM, 0) / total_TPM) %>%
+  ungroup() %>% 
+  filter(transcript_id=='Exon 4')#,
+        # total_TPM > quantile(total_TPM, 0.25))
+
+color_df <- hist_indep_rna_df %>%
+  dplyr::select(plot_group_hex, plot_group) %>%
+  dplyr::filter(!is.na(plot_group)) %>%
+  unique()
+
+cols <- as.character(color_df$plot_group_hex)
+names(cols) <- as.character(color_df$plot_group)
+
+# Ensure no duplicate rows
+color_df <- color_df %>%
+  dplyr::distinct()
+
+# Create the color mapping
+color_mapping <- setNames(color_df$plot_group_hex, color_df$plot_group)
+ 
+# Add a color column to the data based only on plot_group
+transcript_expr_CLK1_combined_df <- transcript_expr_CLK1_combined_df %>%
+  mutate(
+    dot_color = case_when(
+      plot_group %in% names(color_mapping) ~ color_mapping[plot_group], # Use plot_group-specific colors
+      TRUE ~ "#b5b5b5"                                                  # Default to grey
+    )
+  ) %>%
+  filter(!is.na(plot_group)) # Remove rows with NA in plot_group
+
+
+## order evodevo
+evodevo_order <- c(
+  "4 Week Post Conception",
+  "5 Week Post Conception",
+  "6 Week Post Conception",
+  "7 Week Post Conception",
+  "8 Week Post Conception",
+  "9 Week Post Conception",
+  "10 Week Post Conception",
+  "11 Week Post Conception",
+  "12 Week Post Conception",
+  "13 Week Post Conception",
+  "16 Week Post Conception",
+  "Adolescent",
+  "Elderly",
+  "Infant",
+  "Neonate",
+  "School Age Child",
+  "Toddler",
+  "Young Adult",
+  "Middle Adult")
```

Comment:
I can try to update this once all changes are in, it needs a fct_relevel, not a c()